### PR TITLE
Refs #28720 - Fixup migration variables and template

### PR DESCRIPTION
**Pull request by wbclark**
Based on feedback here: https://github.com/theforeman/puppet-pulpcore/pull/63/files

### CODE REVIEW

File: templates/migration-settings.py.erb [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
-    "password": "<%= scope['pulpcore::plugin::migration::mongo_db_password'] %>",
-    "replica_set": "<%= scope['pulpcore::plugin::migration::mongo_db_replica_set'] %>",
-    "ssl": "<%= scope['pulpcore::plugin::migration::mongo_db_ssl'] %>",
-    "ssl_keyfile": "<%= scope['pulpcore::plugin::migration::mongo_db_ssl_keyfile'] %>",
-    "ssl_certfile": "<%= scope['pulpcore::plugin::migration::mongo_db_ssl_certfile'] %>",
-    "verify_ssl": "<%= scope['pulpcore::plugin::migration::mongo_db_verify_ssl'] %>",
-    "ca_path": "<%= scope['pulpcore::plugin::migration::mongo_db_ca_path'] %>",
+    "name": "<%= @mongo_db_name %>",
+    "seeds": "<%= @mongo_db_seeds %>",
+<% if not @mongo_db_username.nil? -%>
```
Comment (ekohl) [inline templates/migration-settings.py.erb]: Perhaps easier to read
```suggestion
<% unless @mongo_db_username.nil? -%>
```

File: templates/migration-settings.py.erb [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+    "username": "<%= @mongo_db_username %>",
+<% end -%>
+<% if not @mongo_db_password.nil? -%>
+    "password": "<%= @mongo_db_password %>",
+<% end -%>
+<% if not @mongo_db_replica_set.nil? -%>
+    "replica_set": "<%= @mongo_db_replica_set %>",
+<% end -%>
+    "ssl": "<%= @mongo_db_ssl ? 'True' : 'False' %>",
```
Comment (ekohl) [inline templates/migration-settings.py.erb]: I think this should actually be a Python boolean
```suggestion
    "ssl": <%= @mongo_db_ssl ? 'True' : 'False' %>,
```

File: templates/migration-settings.py.erb (hunk 1 above)
Comment (wbclark) [reply templates/migration-settings.py.erb]: done.

File: templates/migration-settings.py.erb (hunk 2 above)
Comment (wbclark) [reply templates/migration-settings.py.erb]: and done.

File: spec/classes/plugin_migration_spec.rb [hunk 3]
```diff
@@ -4,12 +4,7 @@
 PULP2_MONGODB = {
     "name": "pulp_database",
     "seeds": "localhost:27017",
-    "username": "",
-    "password": "",
-    "replica_set": "",
     "ssl": "False",
```
Comment (ekohl) [inline spec/classes/plugin_migration_spec.rb]: I would expect this not to be quoted

Comment (wbclark) [reply spec/classes/plugin_migration_spec.rb]: good point

**Pull request merged**